### PR TITLE
Add cmake for aarch64

### PR DESCRIPTION
**Pull request by maxime-desroches**

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Just apt install it? Can we tell poetry to only pick versions with wheels?

File: tools/install_ubuntu_dependencies.sh [hunk 1]
```diff
@@ -18,6 +18,7 @@ function install_ubuntu_common_requirements() {
   $SUDO apt-get install -y --no-install-recommends \
     ca-certificates \
     clang \
+    cmake \
```
[the remark this answers is not in the document]
Comment (maxime-desroches) [reply tools/install_ubuntu_dependencies.sh:21]: The problem is not with cmake anymore, but with the rest of the build.
We hit this error: 

CMake Error at swig/python/CMakeLists.txt:32 (file):
     file STRINGS file "/tmp/tmp5zh0ahys/.venv/include/patchlevel.h" cannot be read.

   Call Stack (most recent call first):
     swig/python/CMakeLists.txt:62 (CasadiGetPythonVersionMajor)


Same problem here: https://github.com/pyodide/pyodide/issues/4551

[the remark this answers is not in the document]
Comment (maxime-desroches) [reply tools/install_ubuntu_dependencies.sh:21]: They seem to have problem with aarch64 build: https://github.com/casadi/casadi/issues/2887

[the remark this answers is not in the document]
Comment (andiradulescu) [reply tools/install_ubuntu_dependencies.sh:21]: I'm trying to fix the casadi issues: https://github.com/andiradulescu/casadi - will submit a PR to casadi as soon as I see the aarch64-py312 build succeed

**Pull request closed without merging**